xonsh/commands_cache.py

```diff
@@ -75,10 +75,20 @@ def remove_dups(p):
                 ret.append(e)
         return ret
 
+    @staticmethod
+    def remove_dups_path(p):
+        ret = list()
+        for e in p:
+            rpe = os.path.realpath(e)
+            e = e if e.lower() == rpe.lower() else rpe
```

Comment:
I don't think lowering paths is supported cross platform

Reply:
Maybe this should be a modification to `remove_dups()`. Is remove_dups used elsewhere?

Reply:
It's only for checking and this line is needed exactly to cross platform - windows support.

Reply:
The call site isn't guarded by an `if ON_WINDOWS` so this would break on Linux and other case-sensitive OSes

Reply:
Hmm... but test fails...

Reply:
OK, I avoided this.